### PR TITLE
Add Login and Restore YourPage

### LINKED ISSUE #51: Navigation for 'your page'

### PR DESCRIPTION
**Pull request by jiangyue12392**
This PR will close #51 and #52

`Login` is added to the navigation bar of all pages.

On `maps.html`, if the user is not yet logged in, the `submit` button will become a `login` button so that when a user clicks on it, he/she will be redirected to login page instead of posting messages.

Your page is also restored to view all messages posted by the user.

Currently, all login redirects to `yourPage` and all logout redirects to `index` page. May take a lot of effort to make it redirect to the page coming from (I am not sure how to do that yet, and might not be that important to change the behaviour).

### CODE REVIEW

File: src/main/webapp/js/maps-page-loader.js [hunk 1]
```diff
@@ -9,6 +9,8 @@ let chairIcon = {
   origin: new google.maps.Point(0,0)
 };
 
+let loginStatus=false;
```
Comment (potay) [inline src/main/webapp/js/maps-page-loader.js]:
```suggestion
let loginStatus = false;
```

File: src/main/webapp/js/maps-page-loader.js [hunk 2]
```diff
@@ -124,15 +126,33 @@ function buildPopupWindowInput(cinemaName, key) {
   document.getElementById('mapsPgPopUp').style.display = "block";
   const textBox = document.getElementById('submitReviewTextArea');
   const button = document.getElementById('submitReviewsButton');
+  if (!loginStatus) {
+    button.innerHTML = "Login"
```
Comment (potay) [inline src/main/webapp/js/maps-page-loader.js]:
```suggestion
    button.innerHTML = "Login to review"
```

File: src/main/webapp/js/maps-page-loader.js [hunk 3]
```diff
@@ -124,15 +126,33 @@ function buildPopupWindowInput(cinemaName, key) {
...
   };
 }
 
 function div_hide() {
   document.getElementById('mapsPgPopUp').style.display = "none";
 }
 
+function fetchLoginStatus() {
+  fetch('/login-status')
+    .then((response) => {
```
Comment (potay) [inline src/main/webapp/js/maps-page-loader.js]:
```suggestion
    .then((response) => response.json())
```

File: src/main/webapp/aboutus.html [hunk 4]
```diff
@@ -25,10 +25,9 @@
   <link rel="stylesheet" type="text/css" href="/css/navbar.css">
   <script type="text/javascript" src='js/main.js'></script>
   <script type = "text/javascript" src="/js/swiper.min.js"></script>
```
Comment (potay) [inline src/main/webapp/aboutus.html]:
```suggestion
  <script type="text/javascript" src="/js/swiper.min.js"></script>
```

**Review by potay — APPROVED** [review]

File: src/main/webapp/js/maps-page-loader.js (hunk 1 above)
Comment (jiangyue12392) [reply src/main/webapp/js/maps-page-loader.js]: Updated

File: src/main/webapp/js/maps-page-loader.js (hunk 2 above)
Comment (jiangyue12392) [reply src/main/webapp/js/maps-page-loader.js]: Updated

File: src/main/webapp/js/maps-page-loader.js (hunk 3 above)
Comment (jiangyue12392) [reply src/main/webapp/js/maps-page-loader.js]: Updated

File: src/main/webapp/aboutus.html (hunk 4 above)
Comment (jiangyue12392) [reply src/main/webapp/aboutus.html]: Updated

**Pull request merged**